### PR TITLE
OF-1528: Allow Java 8 to be used when compiling JSP pages

### PR DESCRIPTION
**Pull request by GregDThomas**

### CODE REVIEW

File: src/web/session-summary.jsp [hunk 1]
```diff
@@ -95,99 +95,72 @@
     sessionResultFilter.setSortOrder(order);
     // Filter out the dodgy looking sessions
     // Note; dodgy looking cast as we're stuck using Java 7 to compile JSPs - see OF-1528
```
Comment (guusdk) [inline src/web/session-summary.jsp]: This comment is no longer valid now, is it?

Comment (GregDThomas) [reply src/web/session-summary.jsp]: You're right, yes.

**Comment by GregDThomas** [comment]
I was just about to merge this when I realised that plugins could not take advantage of Java 8 - so I updated the parent pom.xml for the plugins, too.  As far as I can tell, there's no way to inherit the same jetty version from the core xmppserver pom.xml, though happy to be corrected. 

Also, does this mean we should bump the minJavaVersion to 1.8 for all the plugin.xml files, now?

**Comment by guusdk** [comment]
> I was just about to merge this when I realised that plugins could not take advantage of Java 8 - so I updated the parent pom.xml for the plugins, too.  As far as I can tell, there's no way to inherit the same jetty version from the core xmppserver pom.xml, though happy to be corrected. 

I think you're right

> Also, does this mean we should bump the minJavaVersion to 1.8 for all the plugin.xml files, now?

Unsure - that's property defines a _runtime_ restriction. Does the compiled JSP run on java 7? If not, I guess we need `minJavaVersion` on all plugins that have admin console pages (I'm no sure if all plugins do).

**Comment by GregDThomas** [comment]
A JSP compiled with a target of 1.8 will not run on Java 8.

On reflection, I think the correct route is
a) Remove the <source> / <target> blocks from the parent pom.xml - so by default plugins will compile to Java 7
b) If a plugin requires Java 8 JSP, then can set the <source> / <target> appropriately along with their minJavaVersion.

**Comment by guusdk** [comment]
Maybe this discussion can be short-circuited. I just remembered that on master, we already dropped Java 7 support: https://github.com/igniterealtime/Openfire/pull/1028

**Comment by GregDThomas** [comment]
Good point, well made. I'll leave it as it is.

**Comment by guusdk** [comment]
... do we need to bump the version number for all plugins?

**Comment by GregDThomas** [comment]
Another good point, yes. I'll bump them to SNAPSHOT versions, though - the existing released versions (compiled with Java 7) will continue to work on Openfire 4.3 with Java 8 so there's no need to update them, but next time they are released they will be Java 8 only.

**Comment by GregDThomas** [comment]
(but the good news is it's not all plugins; only those that depend on Openfire 4.3.0-SNAPSHOT)

**Comment by GregDThomas** [comment]
Right, so anything using Openfire 4.3.0 as a parent was bumped to "next-minor-version".SNAPSHOT. Let's hope that's end end of it!

**Comment by akrherz** [comment]
Hi @GregDThomas what's the status of this PR?  Attempting to get things triaged

**Comment by GregDThomas** [comment]
I'll see if I can get it re-based tomorrow

**Comment by GregDThomas** [comment]
Closed in favour of https://github.com/igniterealtime/Openfire/pull/1133

**Pull request closed without merging**